resources/views/hardware/bulk-delete.blade.php

```diff
@@ -42,15 +42,15 @@
                 <td>{{ $asset->id }}</td>
                 <td>{{ $asset->present()->name() }}</td>
                 <td>
-                  @if ($asset->location)
+                  @if ($asset->location_id)
```

Comment:
Why was this line changed? There is a relationship (`$asset->location`) which is being checked for, which is generally better than just checking for the ID, since the ID could be invalid.

Reply:
so, the two things we were checking against are the location_id and the rtd_location_id. these are the two values in the assets table when we looked for location. currently, we are ending up with blanks for the location of an asset on the bulk delete page, this is weird behavior, since it won't match the assets index page. since through `deployedLocationFormatter` we return the `row.name` off the `row.id`, and if not, then we pull the `rtd_location.name` from the `rtd_location.id`. its this if/else that we tried to parity.

Reply:
Right, but if the `location_id` or `rtd_location_id` reference a non-existent location (bad data, etc), that will 500. We should be checking against the relations to make sure those locations actually exist in the database.

Reply:
righto, i'll get to this on monday and get that switched over. ty for the explaination.